### PR TITLE
feat(session): Add alert window when the session is about to expire. …

### LINKED ISSUE #822: Session expires even if user interacts with UI
When working on:
* hawtio/hawtio#3178
* #705

and with short server-side `-Dhawtio.sessionTimeout=120` I found that even if you clean UI every seconds, you're still logged out after 2 minutes.
With default session timeout (1800s = 30 mins) from time time (when clicking) there's a request performed for ... `/hawtio/img/hawtio-logo.svg` - this request in fact is enough to keep the session alive.

### PR DESCRIPTION
**Pull request by grgrzybek**
…Closes #822

* global (fast) onClick handler that sets last user activity timestamp
* <SessionMonitor> React component (added to <HawtioPage>) with session tracking
* alert when the session is about to expire

### CODE REVIEW

**Comment by tadayosi** [comment]
Let's wait until E2E is fixed.

**Review by tadayosi — APPROVED** [review]
Overall, looks great to me, thanks!

There are only a few minor comments on the styling.

File: packages/hawtio/src/ui/session/session-service.ts [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+  shouldResetTimer() {
+    return this.resetTimer
+  }
+
+  clearResetTimerFlag() {
+    this.resetTimer = false
+  }
+
+  async fetchConfiguration() {
```
[the remark this answers is not in the document]
Comment (grgrzybek) [reply packages/hawtio/src/ui/session/session-service.ts:104]: It doesn't return anything, so I added return type of `Promise<void>`.

**Comment by grgrzybek** [comment]
Rebased and pushed.

**Comment by tadayosi** [comment]
please rebase it again.

**Pull request merged**